crates/server/src/admin/handlers/proposals.rs

```diff
@@ -0,0 +1,250 @@
+use crate::AdminState;
+
+//todo split it up into separate files
+
+#[derive(Debug, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub enum ActionType {
+    ExternalFunctionCall,
+    Transfer,
+    SetNumApprovals,
+    SetActiveProposalsLimit,
+    SetContextValue,
+}
+
+#[derive(Debug, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct User {
+    pub(crate) identity_public_key: String,
+}
+
+#[derive(Debug, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub enum Action {
+    ExternalFunctionCall(ExternalFunctionCall),
+    Transfer(Transfer),
+    SetNumApprovals(SetNumApprovals),
+    SetActiveProposalsLimit(SetActiveProposalsLimit),
+    SetContextValue(SetContextValue),
+}
+
+#[derive(Debug, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct ExternalFunctionCall {
+    pub(crate) receiver_id: User,
+    pub(crate) method_name: String,
+    pub(crate) args: Value,
+    pub(crate) deposit: String,
+    pub(crate) gas: String,
+}
+
+#[derive(Debug, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Transfer {
+    pub(crate) amount: String,
+}
+
+#[derive(Debug, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct SetNumApprovals {
+    pub(crate) num_of_approvals: u32,
+}
+
+#[derive(Debug, Deserialize, Serialize)]
+pub struct SetActiveProposalsLimit {
+    pub(crate) active_proposals_limit: u32,
+}
+
+// Use generics to allow any type for `value` in `SetContextValue`
+#[derive(Debug, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct SetContextValue {
+    pub(crate) key: String,
+    pub(crate) value: Value,
+}
+
+// Define Proposal struct
+#[derive(Debug, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Proposal {
+    pub(crate) id: String,
+    pub(crate) author: User,
+    pub(crate) actions: Vec<Action>,
+    pub(crate) title: String,
+    pub(crate) description: String,
+    pub(crate) created_at: String,
+}
+
+// Define Members struct
+#[derive(Debug, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Members {
+    pub public_key: String,
+}
+
+// Define Message struct
+#[derive(Debug, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Message {
+    pub public_key: String,
+}
+
+//ENDPOINTS
+
+#[derive(Debug, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct GetProposalsResponse {
+    data: Vec<Proposal>,
+}
+
+#[derive(Debug, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct GetProposalResponse {
+    data: Proposal,
+}
+
+pub async fn get_proposals_handler(
+    session: Session,
+    Extension(state): Extension<Arc<AdminState>>,
+    Json(req): Json<GetProposalsResponse>,
+) -> impl IntoResponse {
+    let sample_action = Action::ExternalFunctionCall(ExternalFunctionCall {
+        receiver_id: get_mock_user(),
+        method_name: "sampleMethod".to_owned(),
```

Comment:
I think `to_string()` should be everywhere where you are converting a `&str` to `String`, since it's more clear what you are trying to do, but might be just a nitpick, not sure about performance (cc: @miraclx)

Reply:
It was before clippy complaint

Reply:
`str::to_string` [calls](https://github.com/rust-lang/rust/blob/27e38f8fc7efc57b75e9a763d7a0ee44822cd5f7/library/alloc/src/string.rs#L2795) `String::from(&str)` which [calls](https://github.com/rust-lang/rust/blob/27e38f8fc7efc57b75e9a763d7a0ee44822cd5f7/library/alloc/src/string.rs#L2880) `str::to_owned`, all of which are inlined, so in most cases, it should compile to the same code.

I personally prefer just calling `.to_owned()`